decred/decred/dcr/rpc.py

```diff
@@ -12,6 +12,29 @@
 from .wire.msgtx import MsgTx
 
 
+def conv(thing):
+    """
+    Helper method to convert reversed bytes or lists of reversed bytes to hex
+    strings. If it is already a string, it is untouched.
+
+    Args:
+        thing (list(str) or list(ByteArray) or str or ByteArray): The thing to
+            convert to str or a list of str.
+
+    Returs:
+        list(str) or str: The thing converted to a hex encoded str.
+    """
+
+    def c(thing):
+        if isinstance(thing, ByteArray):
+            return reversed(thing).hex()
+        return thing
+
+    if isinstance(thing, list):
```

Comment:
What about tuples, sets, and generators? Most would be covered by `hasattr(thing, "__iter__")`, but so will a `dict`.